konstantinmuenster/strapi-plugin-rich-text
# feat: #17 extend insert link dialog to handle content types

### What does it do?

It adds support for inserting content types' links.

### Why is it needed?

Content Managers want to choose entries from the CMS content types as links in the content.

### How to test it?
Setup the plugin with the content types, the properties regarded as labels, and those regarded as uuid. 

You can also setup these options in the settings panels.
![Screenshot_20241009_191328](https://github.com/user-attachments/assets/a7ecd0f1-4bed-41cc-9f78-fdcc3ba28df5)

Go to the entry with the Rich Text field, insert a link, and see a select input with the available content types.
![Screenshot_20241009_191808](https://github.com/user-attachments/assets/28e2829a-82a0-4270-b77f-291cc2696472)

After choosing the content type you will see another select input with a list of the existing entries, it is an autocomplete input so that you can search the entity.
![Screenshot_20241009_191822](https://github.com/user-attachments/assets/918447e5-9297-4f1f-9dd9-5ff181b80751)

Once you pick the entity, you will see the link source input filled with a calculated URL based on the content type, the singular name, and the entry's uuid field. You can modify it as you like.

### Notes
- It is a basic use case, it can be further improved.
- It requires merging the prior [PR](https://github.com/konstantinmuenster/strapi-plugin-rich-text/pull/28).
